homeassistant/components/roon/config_flow.py

```diff
@@ -0,0 +1,96 @@
+"""Config flow for Roon."""
+from asyncio import sleep
+from collections import OrderedDict
+import logging
+
+from roon import RoonApi
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_KEY, CONF_HOST
+from homeassistant.core import callback
+
+from .const import CONF_CUSTOM_PLAY_ACTION, DOMAIN, ROON_APPINFO
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@callback
+def configured_hosts(hass):
+    """Return a set of the configured hosts."""
+    return {
+        entry.data[CONF_HOST] for entry in hass.config_entries.async_entries(DOMAIN)
+    }
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class FlowHandler(config_entries.ConfigFlow):
+    """Handle a config flow."""
+
+    user_input = {}
+    roonapi = None
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        return await self.async_step_init()
+
+    async def async_step_init(self, user_input=None):
+        """Confirm the setup."""
+        if user_input is not None:
+            return await self.async_step_link(user_input)
```

Comment:
Save the host as an instance attribute before moving to the next step instead. Don't pass it as user_input. Each step may have a form part that will be responsible for the user input to that step.